include/podio/UserDataSIOBlock.h

```diff
@@ -0,0 +1,63 @@
+#ifndef UserDataSIOBlock_H
+#define UserDataSIOBlock_H
+
+#include "podio/SIOBlock.h"
+#include "podio/UserDataCollection.h"
+
+#include <sio/api.h>
+#include <sio/io_device.h>
+#include <sio/version.h>
+
+#include <typeindex>
+#include <string>
+
+
+template <typename BasicType>
+class UserDataSIOBlock: public podio::SIOBlock {
+public:
+  UserDataSIOBlock() :
+    SIOBlock( podio::UserDataTypes::instance().sio_name( typeid(BasicType) ),
+	      sio::version::encode_version(0, 1)) {
+
+    podio::SIOBlockFactory::instance().registerBlockForCollection(
+      podio::UserDataTypes::instance().name( typeid(BasicType) ) , this);
+  }
+
+  UserDataSIOBlock(const std::string& name) :
+    SIOBlock(name, sio::version::encode_version(0, 1)) {}
+
+
+  virtual void read(sio::read_device& device, sio::version_type /*version*/) override {
+    auto collBuffers = _col->getBuffers();
+    if (not _col->isSubsetCollection()) {
```

Comment:
This check could be removed for the `UserDataCollection`, since it will always return true.

Reply:
Indeed.